processing/src/main/java/io/druid/query/groupby/GroupByQuery.java

```diff
@@ -436,6 +444,32 @@ public GroupByQuery withPostAggregatorSpecs(final List<PostAggregator> postAggre
     );
   }
 
+  private static void verifyOutputNames(
+      List<DimensionSpec> dimensions,
+      List<AggregatorFactory> aggregators,
+      List<PostAggregator> postAggregators
+  )
+  {
+    final Set<String> outputNames = Sets.newHashSet();
+    for (DimensionSpec dimension : dimensions) {
+      if (!outputNames.add(dimension.getOutputName())) {
+        throw new IAE("Duplicate output name[%s]", dimension.getOutputName());
+      }
+    }
+
+    for (AggregatorFactory aggregator : aggregators) {
+      if (!outputNames.add(aggregator.getName())) {
+        throw new IAE("Duplicate output name[%s]", aggregator.getName());
```

Comment:
I think that message would be misleading - the duplicate name might be a dimension and aggregator with the same name, not two aggregators.